File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/state/StatePattern.java
package jv.com.tencent.cain.design_pattern.state;

/**
 * @author cainjiang
 * @date 2020/3/27
 * 状态设计模式
 */
public class StatePattern {
    public static void main(String[] args) {
        Work emergencyProject = new Work();
        emergencyProject.setHour(9);
        emergencyProject.doWork();
        emergencyProject.setHour(10);
        emergencyProject.doWork();
        emergencyProject.setHour(15);
        emergencyProject.doWork();
        emergencyProject.setFinished(false);
        emergencyProject.setHour(19);
        emergencyProject.doWork();
        emergencyProject.setFinished(true);
        emergencyProject.doWork();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/state/StatePattern.kt
package kt.com.tencent.cain.desigin_pattern.state

/**
 * @author : jiangyu
 * @date   : 2020/3/28
 * @desc   : 状态设计模式
 */

fun main() {
    val work = Work()
    work.curTimeHour = 10
    work.doWork()
    work.curTimeHour = 13
    work.doWork()
    work.curTimeHour = 15
    work.doWork()
    work.curTimeHour = 20
    work.doWork()
    work.isFinished = true
    work.doWork()
}